File: frontend-admin/lib/api/storeTemplateBinding.ts
import { api } from "./client";

/**
 * 店铺-模板版本绑定 API（Track AS5）。
 *
 * <p>一店一行；用于跨店推送时拉取该店应用的占位符替换规则。
 * GET 返回 null 表示未绑定（前端渲染空态 / 兜底用模板默认）。
 */
export type StoreTemplateBinding = {
  id: number;
  storeId: number;
  baseTemplateVersionId: number;
  customReplaceRulesJson: string | null;
  boundAt: string;
  boundBy: number | null;
};

export const storeTemplateBindingApi = {
  /** 拿当前 binding；后端返回 null 表示该店未绑定。 */
  get: (storeId: number) =>
    api.get<StoreTemplateBinding | null>(`/store-template-binding/${storeId}`),

  /** 绑定/重绑该店到指定 base_template_version。 */
  upsert: (
    storeId: number,
    body: {
      baseTemplateVersionId: number;
      /** 可选自定义规则 JSON 字符串（同 key 覆盖模板默认） */
      customReplaceRulesJson?: string | null;
    },
  ) => api.put<void>(`/store-template-binding/${storeId}`, body),

  /** 解绑（删除该行）。 */
  remove: (storeId: number) =>
    api.del<void>(`/store-template-binding/${storeId}`),
};

File: backend-api/src/main/java/com/biou/shopifyhub/template/binding/StoreTemplateBinding.java
package com.biou.shopifyhub.template.binding;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.time.LocalDateTime;

/**
 * 店铺与基础模板版本的绑定（Track AS5）。一店一行，唯一键 {@code uk_store(store_id)}。
 *
 * <p>{@code base_template_version_id} 指向 {@code base_template_version.id}，决定推送
 * 时使用哪条 {@code default_replace_rules_json}。{@code custom_replace_rules_json}
 * 是该店额外覆盖/扩展的 JSON（同 key 覆盖默认）。
 */
@TableName("store_template_binding")
public class StoreTemplateBinding {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long storeId;
    private Long baseTemplateVersionId;
    private String customReplaceRulesJson;
    private LocalDateTime boundAt;
    private Long boundBy;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getStoreId() { return storeId; }
    public void setStoreId(Long storeId) { this.storeId = storeId; }
    public Long getBaseTemplateVersionId() { return baseTemplateVersionId; }
    public void setBaseTemplateVersionId(Long baseTemplateVersionId) { this.baseTemplateVersionId = baseTemplateVersionId; }
    public String getCustomReplaceRulesJson() { return customReplaceRulesJson; }
    public void setCustomReplaceRulesJson(String customReplaceRulesJson) { this.customReplaceRulesJson = customReplaceRulesJson; }
    public LocalDateTime getBoundAt() { return boundAt; }
    public void setBoundAt(LocalDateTime boundAt) { this.boundAt = boundAt; }
    public Long getBoundBy() { return boundBy; }
    public void setBoundBy(Long boundBy) { this.boundBy = boundBy; }
}
